DRODLib/Character.cpp

```diff
@@ -5093,6 +5214,57 @@ void CCharacter::SetVariable(const CCharacterCommand& command, CCurrentGame* pGa
 	}
 }
 
+//*****************************************************************************
+void CCharacter::SetArrayVariable(
+	const CCharacterCommand& command,
+	CCurrentGame* pGame,
+	CCueEvents& CueEvents)
+{
+	const UINT varIndex = command.w;
+	if (pGame->pHold && !pGame->pHold->IsArrayVar(varIndex))
+		return; //Don't set normal var as an array var
+
+	int arrayIndex = (int)(this->paramF == NO_OVERRIDE ? command.flags : this->paramF);
+	ScriptArrayMap& scriptArrays = pGame->pHold->IsLocalVar(varIndex) ? this->localScriptArrays : pGame->scriptArrays;
+
+	vector<WSTRING> expressions = WCSExplode(command.label, *wszSemicolon);
+	for (vector<WSTRING>::const_iterator expression = expressions.begin();
+		expression != expressions.end(); ++expression) {
+		UINT index = 0;
+		int x = scriptArrays[varIndex][arrayIndex];
+		int operand = 0;
+		operand = parseExpression(expression->c_str(), index, pGame, this);
+
+		switch (command.h)
+		{
+			case ScriptVars::Assign:
+				x = operand;
+			break;
+			case ScriptVars::Inc:
+				addWithClamp(x, operand);
+			break;
+			case ScriptVars::Dec:
+				addWithClamp(x, -operand);
+			break;
+			case ScriptVars::MultiplyBy:
+				multWithClamp(x, operand);
+			break;
+			case ScriptVars::DivideBy:
+				if (operand)
+					x /= operand;
+			break;
+			case ScriptVars::Mod:
+				if (operand)
+					x = x % operand;
+			break;
+			default: break;
+		}
+
+		scriptArrays[varIndex][arrayIndex] = x;
```

Comment:
I'm a bit worried about this bloating memory needlessly when the set value is zero. How about if x is 0, then unset the map at arrayIndex to manage memory?